src/prpy/util.py

```diff
@@ -1326,7 +1326,7 @@ def ComputeGeodesicUnitTiming(traj, env=None, alpha=1.0):
     return new_traj
 
 
-def CheckJointLimits(robot, q, deterministic=None):
+def CheckJointLimits(robot, q, hard_limits, deterministic=None):
```

Comment:
Well the hard limits are indeed robot specific, which is why I thought to send them as an argument, but can we just make them a property of the robot then?
As for the planner's awareness to limits, I figured to use the `ompl` planners for example, where `or_ompl` enforces the joint limits, we would anyway have to change that logic to accommodate both kinds of limits right? Since `or_ompl` planners do not interact with `prpy.util`